examples/otlp-push-controller/src/main.rs

```diff
@@ -0,0 +1,51 @@
+use autometrics::autometrics;
+use opentelemetry::{runtime, Context};
+use opentelemetry::sdk::export::metrics::aggregation::cumulative_temporality_selector;
+use opentelemetry::sdk::metrics::controllers::BasicController;
+use opentelemetry::sdk::metrics::selectors;
+use opentelemetry::metrics;
+use opentelemetry_otlp::{ExportConfig, WithExportConfig};
+use tokio::time::sleep;
+use std::error::Error;
+use std::time::Duration;
+
+fn init_metrics() -> metrics::Result<BasicController> {
+    let export_config = ExportConfig {
+        endpoint: "http://localhost:4317".to_string(),
+        ..ExportConfig::default()
+    };
+    opentelemetry_otlp::new_pipeline()
+        .metrics(
+            selectors::simple::inexpensive(),
+            cumulative_temporality_selector(),
+            runtime::Tokio,
+        )
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .tonic()
+                .with_export_config(export_config),
+        )
+        .build()
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let meter_provider = init_metrics()?;
+    let cx = Context::current();
+
+    for numb in 0..10 {
+        do_stuff(numb).await;
+    }
+
+    println!("Waiting so that we could see metrics being pushed via OTLP every 10 seconds...");
+    sleep(Duration::from_secs(60)).await;
+    meter_provider.stop(&cx)?;
+
+    Ok(())
+}
+
+#[autometrics]
+async fn do_stuff(numb: u64) {
+    println!("Doing stuff for {} seconds...", numb);
+    sleep(Duration::from_secs(numb)).await;
```

Comment:
I see, so it'll only have a single push to the collector? I could go either way on that, so feel free to leave it in if you think it's better that way

Reply:
Hey there, sorry for disappearing for a few days - work got in the way. I’m leaning towards the solution that simulates load and has more pushes to the metric collector. I’m planning to tidy up the code a bit and make it look nicer, hopefully by the end of the week.

Reply:
Sounds good. Thanks for the update!

Reply:
Hi @emschwartz ,
I have pushed a new commit in which I have revised the code so that the example completes in about 30 seconds and allows for a series of realistic metrics to be seen.

let me know what you think about!